src/bidiMapper/domains/events/EventManager.ts

```diff
@@ -86,4 +144,89 @@ export class EventManager implements IEventManager {
       }
     }
   }
+
+  /**
+   * If the event is buffer-able, put it in the buffer.
+   * @param eventWrapper
+   * @param contextId
+   * @private
+   */
+  #bufferEvent(
+    eventWrapper: IdWrapper<Message.EventMessage>,
+    contextId: CommonDataTypes.BrowsingContext | null
+  ) {
+    const eventName = eventWrapper.value.method;
+    if (!EventManager.#eventBufferLength.has(eventName)) {
+      // Do nothing if the event is no buffer-able.
+      return;
+    }
+    const bufferMapKey = EventManager.#getMapKey(eventName, contextId);
+    if (!this.#eventBuffers.has(bufferMapKey)) {
+      this.#eventBuffers.set(
+        bufferMapKey,
+        new Buffer<IdWrapper<Message.EventMessage>>(
+          EventManager.#eventBufferLength.get(eventName)!
+        )
+      );
+    }
+    this.#eventBuffers.get(bufferMapKey)!.add(eventWrapper);
+  }
+
+  /**
+   * If the event is buffer-able, mark it as sent to the given contextId and channel.
+   * @param eventWrapper
+   * @param contextId
+   * @param channel
+   * @private
+   */
+  #markEventSent(
+    eventWrapper: IdWrapper<Message.EventMessage>,
+    contextId: CommonDataTypes.BrowsingContext | null,
+    channel: string | null
+  ) {
+    const eventName = eventWrapper.value.method;
+    if (!EventManager.#eventBufferLength.has(eventName)) {
+      // Do nothing if the event is no buffer-able.
+      return;
+    }
+
+    const lastSentMapKey = EventManager.#getMapKey(
+      eventName,
+      contextId,
+      channel
+    );
+    this.#lastMessageSent.set(
+      lastSentMapKey,
+      Math.max(this.#lastMessageSent.get(lastSentMapKey) ?? 0, eventWrapper.id)
+    );
+  }
+
+  /**
+   * Returns events which are buffered and not yet sent to the given channel events.
+   * @param eventName
+   * @param contextId
+   * @param channel
+   * @private
```

Comment:
What is the point of the `@private` annotation, given that the function is marked with `#` already?

Reply:
It is auto-generated by IntelliJ.

Reply:
Then we should remove them. In fact, we can even remove the `@param`s. It's redundant jsdoc noise.

Reply:
Removed in this file, but I'd reduce the amount of manual corrections to as low as possible.